distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Excel_Writer.enso

```diff
@@ -34,7 +34,8 @@ make_java_existing_data_mode on_existing_file match_columns = case on_existing_f
 write_file : File -> Table -> Existing_File_Behavior -> Excel_Section -> (Boolean|Infer) -> Match_Columns -> Problem_Behavior -> Boolean -> File
 write_file file table on_existing_file section headers match_columns on_problems xls_format=False =
     _ = [on_problems]
-    workbook = if file.exists.not then ExcelWriter.createWorkbook xls_format else
+    ## If file does not exist or is empty then create a new workbook.
+    workbook = if file.exists.not || (file.size == 0) then ExcelWriter.createWorkbook xls_format else
```

Comment:
This looks good, but I'd like to understand out of curiosity.

I assume the situation was encountered in just Enso, without external programs creating the empty file, right?
What scenario was causing an _empty Excel file_ to be created?

This fix looks good and should be there, since other programs may create such empty file too and we want it to work. 

But I wonder if we can avoid creating such an empty file in Enso in the first place?

Reply:
btw. `Arguments:` section in the docstring above is empty. We must have overlooked it in the past. If updating this function, it could be worth to update it as well, or if preferred, just remove the Arguments section altogether, but let's not keep an empty one.

Reply:
Calling `createTempFile` in Java creates a 0 byte file. It felt the correct thing to ignore any 0 byte file here as they would never be an Excel file. Given the want to ensure the temporary files are cleaned up appropriate by the runtime I did not want to delete the empty file as then it might end up not being tracked correctly on close.

Reply:
Ok, I see, Thanks for the explanation